clients/src/main/java/org/apache/kafka/common/network/Selector.java

```diff
@@ -1061,6 +1127,37 @@ public void close() {
         }
     }
 
+    private static class DelayedChannelClose {
+        private final Time time;
+        private final KafkaChannel channel;
+        private final CloseMode closeMode;
+        private final long endTime;
+
+        public DelayedChannelClose(Time time, long startMs, int minimumDelayMs, KafkaChannel channel, CloseMode closeMode) {
+            this.time = time;
+            this.channel = channel;
+            this.closeMode = closeMode;
+            this.endTime = computeDelay(time, startMs, minimumDelayMs);
+        }
+
+        private long computeDelay(Time time, long startMs, int minimumDelayMs) {
+            if (startMs < 0)
+                throw new IllegalArgumentException();
+
+            long now = time.milliseconds();
```

Comment:
Time should get propagated from `Selector`. We should avoid extra calls to `System.currentTimeMillis()`